ota-plus-web/docs/modules/ROOT/pages/use-case-ecm.adoc

```diff
@@ -0,0 +1,287 @@
+. Enter the name like "__Tireon-2018-Scandinavia-010A__" for the group name.
+. In the *Optional Import* section, click *Choose file* and select the "__Tireon-vins.txt__" file that you saved previously.
+. Click **Create**.
++
+You should see matching devices show up in your group. For simplicity's sake, let's say that the group contains exactly 1,000 devices. This is useful to know when we track the progress of our software updates.
+
+== Preparing the software update
+
+// MC: Hack to make this section collapsible
++++ <details><summary> +++
+*What should have happened first* 
++++ </summary><div> +++
+
+Your developers have bitbaked a "baseline" disk image - a baseline disk image that runs application software which is compatible with your ECU's hardware. In this case, it's "NXP-AUTOSAR-Classic-4.2".
+
+The baseline disk image includes version 5.2.0 of the "Engine Idle Shutdown Timer Calibration" software and installed it on your devices - so your cars are running version 5.2.0 right now.
+
+To prepare the new update, your developers have:
+
+* built a version of the "baseline" disk image
+* bumped the minor version to V1.2.2
+** This is because the "Engine Idle Shutdown Timer Calibration" software is the only thing they changed on the disk image.
++
+In other words, it's a minor release.
+
++++ </div></details> +++
+
+As part of the build process the disk image is automatically uploaded to your OTA Connect software repository. Your developers have kindly left the following internal release note.
+
+.Release Note
+====
+Update for NXP S32S24 K ECMs on post-2018 Tireon models.
+
+CHANGED PACKAGES:
+
+* Engine Idle Shutdown Timer Calibration: Updated from 5.2.0 to 5.3.0. 
+** 5.3.0: Overrides engine shutdown if temperature falls below 7.2 degrees C or above 35 degrees C.
+====
+
+So you know that you need version 1.2.2 of the disk image for NXP S32S24 microcontrollers that are used as engine control modules on all Lannista "Tireon" models produced after 2018.
+
+*Create the software update:*
+
+. In the OTA Connect portal, navigate to the *Software updates* page.
+. Click *Create software update* 
+. On the *Details* page, you'd enter something like this:
++
+.Software update name
+====
+Update NXP S32S24 ECMs to improve HVAC in extreme conditions- 
+====
++
+.Software update description
+====
+Update Engine Idle Shutdown Timer Calibration on NXP MPC5777C ECMs in "Tireon" models produced after 2018. 
+
+* Improve HVAC in extreme weather conditions (Initially for Scandinavian market).
+====
+. In the *Control Units* section, you would select the "__ECM (NXP S32V23-M4)__".
++
+** In the *SOFTWARE CURRENTLY INSTALLED* column and the *SOFTWARE TO BE INSTALLED* columns, you would select "Engine Idle Shutdown Timer Calibration" from the dropdown.
+** In the *VERSION CURRENTLY INSTALLED* column, you would select "5.2.0" from the dropdown and in the *VERSION TO BE INSTALLED* column you would select "5.3.0".
```

Comment:
The version here would be the image version. We don't install individual software packages on ECUs, just images. The only time the 5.3.0 would appear would be in the release notes, if at all. More realistic would be that there isn't a separate version number for this functionality.

Reply:
@tkfu Ok, just to help me understand better - when I use a system like Ubuntu, and I am installing packages, I often get warnings like the "`You are currently using version 2.1 of `lib-foo` but  `lib-bar` requires version 2.2 of `lib-foo`" - Doesn't the whole ecosystem of Linux require packages to be explicitly versioned? 
If you fixed some calibration software, wouldn't you want to know the explicit version of the package that contains the fix *as well as* the overall image that contains the fixed package? also - couldn't multiple image versions/variants contain the fixed package? (i.e. `lib-foo` v2.2)

Reply:
> Doesn't the whole ecosystem of Linux require packages to be explicitly versioned?

You wouldn't be running Linux on an ECU like this; if you're running an operating system at all, it'll be some very minimal real-time OS, almost certainly without dynamically linked libraries. You also most likely wouldn't have a separate named package for a function like this. There might be a file somewhere in your source tree called, say, engine_shutdown_conditions_controller.c or something like that, but that'd be the extent of it.

> If you fixed some calibration software, wouldn't you want to know the explicit version of the package that contains the fix as well as the overall image that contains the fixed package? 

If the target was a fairly complex secondary that was meaningfully composed of packages, then yes, arguably, it would be useful to know about that. However, we don't currently have any support for storing that kind of information about secondary ECUs.

> also - couldn't multiple image versions/variants contain the fixed package? (i.e. lib-foo v2.2)

Sure, yes. But the real way this would be managed is by rebuilding all of the image variants to include the fix, and putting that in the release notes.

Reply:
OK , decided to use a different board model/software (Renesas RL78/15, FreeRTOS kernel) in the examples.

Reply:
OK version selection hints fixed in 1aefef24a0c105565faca2b2af72e3ba034a25dc